cluster/gce/gci/shutdown.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A script that let's gci preemptible nodes gracefully terminate in the event of a VM shutdown.
+preemptible=$(curl "http://metadata.google.internal/computeMetadata/v1/instance/scheduling/preemptible" -H "Metadata-Flavor: Google")
+if [ ${preemptible} == "TRUE" ]; then
+    echo "Shutting down! Sleeping for a minute to let the node gracefully terminate"
+    # https://cloud.google.com/compute/docs/instances/stopping-or-deleting-an-instance#delete_timeout
+    sleep 60
```

Comment:
For preemptible instances, it says the period is 30s.

It also says: "If you choose to run a shutdown script during this period, your shutdown script must finish running within this time period so that the operating system has time to complete its shutdown and flush buffers to disk."

So, maybe we should set this to a little less than 30s?

Reply:
The limit doesn't matter much in reality. The VM will be brought down no matter what. That comment is saying that shutdown scripts have 30 seconds to do any smarts. In this case we are not doing anything smart in the shutdown script.

Reply:
Maybe just make it 30s to be consistent?

Reply:
Done